### PR TITLE
camerad: fix memory leak in create_thumbnail

### PR DESCRIPTION
**Pull request by deanlee**
Related issue: https://github.com/commaai/openpilot/issues/2739

I'm not sure this issue can be completely resolved after this fix.need to continue to look for possible memory leaks elsewhere

### CODE REVIEW

**Comment by deanlee** [comment]
the memory returned by jpeg_mem_dest should be freed too

File: selfdrive/camerad/cameras/camera_common.cc [hunk 1]
```diff
@@ -294,15 +294,18 @@ void create_thumbnail(MultiCameraState *s, CameraState *c, uint8_t *bgr_ptr) {
...
+  jpeg_destroy_compress(&cinfo);
+  free(row);
 
   MessageBuilder msg;
   auto thumbnaild = msg.initEvent().initThumbnail();
   thumbnaild.setFrameId(b->cur_frame_data.frame_id);
   thumbnaild.setTimestampEof(b->cur_frame_data.timestamp_eof);
   thumbnaild.setThumbnail(kj::arrayPtr((const uint8_t*)thumbnail_buffer, thumbnail_len));
 
+  free(thumbnail_buffer);
```
Comment (pd0wm) [inline selfdrive/camerad/cameras/camera_common.cc:307]: Can we free before sending the message? Does the `setThumbnail` call make a copy or just reference the `thumbnail_buffer`?

Comment (deanlee) [reply selfdrive/camerad/cameras/camera_common.cc:307]: my bad

**Comment by deanlee** [comment]
my mind went blank.I should free thumbnail_buffer after send.

**Comment by pd0wm** [comment]
No problem. Thanks for finding the leak!

I'll add the jpeg processing to `camera_frame_stream` so we can catch this on the computer as well. Also confirmed the existence of this (obvious) leak using valgrind just as a sanity check and made sure it was gone after your changes.

```
==1759538== 73,884 (1,176 direct, 72,708 indirect) bytes in 7 blocks are definitely lost in loss record 189 of 192
==1759538==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==1759538==    by 0x4A11D1B: jinit_memory_mgr (in /usr/lib/x86_64-linux-gnu/libjpeg.so.8.2.2)
==1759538==    by 0x49E56AC: jpeg_CreateCompress (in /usr/lib/x86_64-linux-gnu/libjpeg.so.8.2.2)
==1759538==    by 0x407F30: create_thumbnail(MultiCameraState*, CameraState*, unsigned char*) (camera_common.cc:262)
==1759538==    by 0x408730: processing_thread(MultiCameraState*, char const*, CameraState*, int, void (*)(MultiCameraState*, CameraState*, int)) (camera_common.cc:360)
==1759538==    by 0x4F9FD83: ??? (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.28)
==1759538==    by 0x49C7608: start_thread (pthread_create.c:477)
==1759538==    by 0x51E9292: clone (clone.S:95)
```

**Pull request merged**